File: integration-test/java/spring-app/src/main/java/com/qualityminds/lazyval/integration/boundary/persistence/jpa/SpringDataRepository.java
package com.qualityminds.lazyval.integration.boundary.persistence.jpa;

import com.qualityminds.lazyval.integration.shared.Isbn;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.CrudRepository;
import org.springframework.stereotype.Repository;

import java.util.List;
import java.util.UUID;

@Repository
public interface SpringDataRepository extends CrudRepository<JpaOrder, UUID> {

    @Query("SELECT o FROM JpaOrder o WHERE o.isbn = :isbn")
    List<JpaOrder> findByIsbn(Isbn isbn);
}

File: integration-test/kotlin/spring-app/src/main/kotlin/com/qualityminds/lazyval/integration/boundary/persistence/jpa/SpringDataRepository.kt
package com.qualityminds.lazyval.integration.boundary.persistence.jpa

import com.qualityminds.lazyval.integration.shared.Isbn
import org.springframework.data.jpa.repository.Query
import org.springframework.data.repository.CrudRepository
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface SpringDataRepository : CrudRepository<JpaOrder, UUID> {

    @Query("SELECT o FROM JpaOrder o WHERE o.isbn = :isbn")
    fun findByIsbn(isbn: Isbn): List<JpaOrder>
}